### PR TITLE
Dev pa auth

### PR DESCRIPTION
**Pull request by racine-p-a**
Inserting all user stuff with tests and guards + adding first implementation from @Pipazoul about projects.

### CODE REVIEW

File: apps/backend/cypress/e2e/user.cy.ts [hunk 1]
```diff
@@ -1,74 +1,253 @@
...
+    password: 'user_test_pw',
+    role: 'ADMIN',
+    active: true,
   };
-  let id_test_user = null;
+  let jwtEditorUser;
+  let idEditorUser;
+  let jwtAdminUser;
+  let idAdminUser;
+  let idUserTestEditor;
```
Comment (Pipazoul) [inline apps/backend/cypress/e2e/user.cy.ts:27]: Github action seems to have noticed some unused vars

`'idAdminUser' is assigned a value but never used`

`'idUserTestAdmin' is assigned a value but never used`

**Review by Pipazoul — APPROVED** [review]

File: apps/backend/cypress/e2e/user.cy.ts (hunk 1 above)
Comment (racine-p-a) [reply apps/backend/cypress/e2e/user.cy.ts:27]: Done.

File: README.md [hunk 2]
```diff
@@ -1,25 +1,7 @@
 # Datatlas
 
-## Getting started
```
Comment (lutangar) [inline README.md]: Verify if on purpose.

File: apps/backend/cypress/e2e/auth.cy.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    }).then((response) => {
+      jwtUser = response.body.access_token;
+      idUserAdmin = response.body.user_id;
+      expect(response.status).to.eq(201);
+    });
+  });
+  it('Auth -> Check self-profile of admin user with wrong jwt.', () => {
+    cy.request({
+      method: 'GET',
+      url: '/api/user/' + idUserAdmin,
```
Comment (lutangar) [inline apps/backend/cypress/e2e/auth.cy.ts]:
```suggestion
      url: `/api/user/${idUserAdmin}`,
```

File: apps/backend/cypress/e2e/auth.cy.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        bearer: 'incorrect_jwt',
+      },
+    }).then((response) => {
+      expect(response.status).to.eq(403);
+    });
+  });
+  it('Auth -> Check self-profile of admin user with correct jwt.', () => {
+    cy.request({
+      method: 'GET',
+      url: '/api/user/' + idUserAdmin,
```
Comment (lutangar) [inline apps/backend/cypress/e2e/auth.cy.ts]:
```suggestion
      url: `/api/user/${idUserAdmin}`,
```

File: apps/backend/src/app/app.controller.ts [hunk 5]
```diff
@@ -10,14 +9,17 @@ export class AppController {
...
+    const test = await this.authService.login(req.body);
+    Logger.log(test);
+    return test;
   }
-
+  /*
   @UseGuards(JwtAuthGuard)
   @Get('profile')
   getProfile(@Request() req) {
+    console.log('début route profile');
```
Comment (lutangar) [inline apps/backend/src/app/app.controller.ts]: Leftover `console.log`

File: apps/backend/src/auth/admin.guard.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+@Injectable()
+export class AdminGuard extends AuthGuard('local') {
+  constructor(private readonly _reflector: Reflector, private jwtService: JwtService) {
+    super();
+  }
+
+  canActivate(context: ExecutionContext) {
+    /*
+        CHECKING IS ADMIN ? -> USE THE JWT
```
Comment (lutangar) [inline apps/backend/src/auth/admin.guard.ts:15]: Let's talk about that !

File: apps/frontend/src/hooks/useTimeElapsed.ts [hunk 7]
```diff
@@ -4,7 +4,7 @@ import { DateTime } from 'luxon';
 // 2 options with luxon to display a human state modified time. toRelative to have a more specific time of modified, or toRelativeCalendar to have a friendly human value.
 
 const useTimeElapsed = (createdAt) => {
-  const [timeElapsed, setTimeElapsed] = useState<string | null>(null);
+  const [timeElapsed, setTimeElapsed] = useState(0);
```
Comment (lutangar) [inline apps/frontend/src/hooks/useTimeElapsed.ts]: This diff mustn't appears in this PR.

File: apps/frontend/src/index.tsx [hunk 8]
```diff
@@ -44,11 +44,7 @@ const router = createBrowserRouter([
 ]);
 
 // Start the mocking conditionally.
-if (
-  process.env.NODE_ENV === 'test' ||
-  process.env.NODE_ENV === 'development' ||
-  process.env.NODE_ENV === 'production'
-) {
+if (process.env.NODE_ENV === 'test' || process.env.NODE_ENV === 'development') {
```
Comment (lutangar) [inline apps/frontend/src/index.tsx]: Mustn't be changed yet.

**Pull request merged**